plugins/inputs/ras/ras.go

```diff
@@ -77,20 +80,39 @@ func (r *Ras) Description() string {
 	return "RAS plugin exposes counter metrics for Machine Check Errors provided by RASDaemon (sqlite3 output is required)."
 }
 
-// Gather reads the stats provided by RASDaemon and writes it to the Accumulator.
-func (r *Ras) Gather(acc telegraf.Accumulator) error {
+// Start initializes connection to DB, metrics are gathered in Gather
+func (r *Ras) Start(telegraf.Accumulator) error {
 	err := validateDbPath(r.DBPath)
 	if err != nil {
 		return err
 	}
 
-	db, err := connectToDB(r.DBPath)
+	r.db, err = connectToDB(r.DBPath)
 	if err != nil {
 		return err
 	}
-	defer db.Close()
 
-	rows, err := db.Query(mceQuery, r.latestTimestamp)
+	return nil
+}
+
+// Stop closes any existing DB connection
+func (r *Ras) Stop() {
+	if r.db != nil {
+		err := r.db.Close()
+		if err != nil {
+			r.Log.Errorf("Error appeared during closing DB (%s): %v", r.DBPath, err)
+		}
+	}
+}
+
+// Gather reads the stats provided by RASDaemon and writes it to the Accumulator.
+func (r *Ras) Gather(acc telegraf.Accumulator) error {
+	err := r.db.Ping()
+	if err != nil {
+		return fmt.Errorf("connection to DB (%s) couldn't be established: %v", r.DBPath, err)
+	}
```

Comment:
I'd consider skipping this and just doing the db.Query. If it's possible to detect the failed connection error and try reconnecting that would be fine.

Reply:
You're right! `db.Query -> db.QueryContext -> db.query` do the same potential reconnect as `db.Ping` so even direct reconnection is not needed.